r/R/dplyr-arrange.R

```diff
@@ -77,6 +77,10 @@ find_and_remove_desc <- function(quosure) {
       # remove enclosing parentheses
       expr <- expr[[2]]
     } else if (identical(expr[[1]], quote(desc))) {
+      # ensure desc() has only one argument
+      if (length(expr) > 2) {
+        stop("desc() expects only one argument", call. = FALSE)
+      }
```

Comment:
This is less of a comment, but more that I want to make sure I'm understanding how this all happens: This _does not_ trigger pull-into-r-and-run-with-dplyr, because [when it is called](https://github.com/apache/arrow/blob/master/r/R/dplyr-arrange.R#L40) is outside fo the catch for try-errors a few lines down, and so this will error + not attempt to (re) run it in R as a result. Do I have that right?

Reply:
Yes, that's correct 👍

Reply:
Also, because this might be confusing: when an R expression is a function call, having length == 2 means it has one and only one argument.

Reply:
Oh yeah, I didn't think anything odd of that cause I've had to parse calls like that before, but maybe it's worth a comment cause that is a bit weird now that you point it out

Reply:
Comment added in ae132c6

Reply:
dplyr doesn't error on this (and also appears to ignore the second argument): 

```r
mtcars %>% arrange(desc(cyl, hp))
```

Notably `mtcars %>% arrange(desc(cyl, asdfasdf))` does not error.

Seems like something that should be reported upstream too--can you do that?

Reply:
Ah I see you already have